### PR TITLE
add whitespace validation for 1995 stem fields

### PR DESCRIPTION
**Pull request by hughes-dustin**
## Description
https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/vets.gov-team/18655

## Screenshots
Screen Shot 2019-08-19 at 10 07 28 AM

Screen Shot 2019-08-19 at 10 08 24 AM

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by zurbergram — CHANGES_REQUESTED** [review]
add unit tests in `vets-website/src/platform/forms/tests/` for the new validations

File: src/platform/forms/validations.js [hunk 1]
```diff
@@ -348,6 +348,20 @@ function validateLength(
   };
 }
 
+function validateWhiteSpace(errors, input) {
+  if (typeof input !== 'undefined') {
+    if (
+      input.toString().startsWith(' ') ||
```
Comment (shawkey-daniel) [inline src/platform/forms/validations.js]: This checks for space, but I was thinking a regex for all whitespace like [^\s]

**Review by shawkey-daniel — CHANGES_REQUESTED** [review]
Use regex to check for any whitespace

File: src/platform/forms/validations.js (hunk 1 above)
Comment (hughes-dustin) [reply src/platform/forms/validations.js]: I figured there were cases where people have first names that include spaces. For instance: "Mary Jane".

Comment (shawkey-daniel) [reply src/platform/forms/validations.js]: That's fine. The ^ is for starts with.

Comment (shawkey-daniel) [reply src/platform/forms/validations.js]: \s is for any whitespace character

File: src/applications/edu-benefits/1995/config/chapters.js [hunk 2]
```diff
@@ -17,6 +17,7 @@ import { showSchoolAddress } from '../../utils/helpers';
 import { displayActiveDutyStem } from '../helpers';
 
 import { activeDuty, benefitSelection, stem } from '../pages';
+import { validateWhiteSpace } from '../../../../platform/forms/validations';
```
Comment (shawkey-daniel) [inline src/applications/edu-benefits/1995/config/chapters.js]: You should be able to just use the relative link /platform/forms/validations

**Review by shawkey-daniel — CHANGES_REQUESTED** [review]
Can we try to find the right regex for starts with any whitespace and end with any whitespace?

File: src/applications/edu-benefits/1995/config/chapters.js (hunk 2 above)
Comment (hughes-dustin) [reply src/applications/edu-benefits/1995/config/chapters.js]: Relative link does not work.

Comment (shawkey-daniel) [reply src/applications/edu-benefits/1995/config/chapters.js]: My bad, platform/forms/validations

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Review by zurbergram — APPROVED** [review]
LGTM

**Review by delli-gatti-michael — APPROVED** [review]
Passes QA

**Comment by hughes-dustin** [comment]
@department-of-veterans-affairs/frontend-review-group Could I get a code review on this at your earliest convenience. Thank you!

File: src/platform/forms/validations.js [hunk 3]
```diff
@@ -348,6 +348,14 @@ function validateLength(
   };
 }
 
+function validateWhiteSpace(errors, input) {
```
Comment (cvalarida) [inline src/platform/forms/validations.js]: I was trying to find a better name for this, but came up with nothing. :man_shrugging:

Can you add a little comment above this saying that it ensures input isn't _only_ whitespace?

**Review by cvalarida — APPROVED** [review]
Please add that comment before merging, but it otherwise looks great. :+1: :shipit:

**Pull request merged**